src/os/bluestore/BlueFS.cc

```diff
@@ -1136,13 +1136,15 @@ void BlueFS::_compact_log_sync()
   log_writer->append(bl);
   int r = _flush(log_writer, true);
   assert(r == 0);
-  wait_for_aio(log_writer);
+  _flush_bdev_safely(log_writer);
```

Comment:
This is a subtle change because the lock is now being dropped before the super.log_fnode is updated.